app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala

```diff
@@ -187,33 +208,81 @@ class SubscriptionController @Inject()(
             sessionStoreService.fetchInitialDetails.flatMap { maybeInitialDetails =>
               maybeInitialDetails
                 .map { initialDetails =>
-                  subscriptionService.subscribe(initialDetails, validDesAddress).map(redirectSubscriptionResponse)
+                  subscriptionService.subscribe(initialDetails, validDesAddress).flatMap(redirectSubscriptionResponse)
                 }
                 .getOrElse(Future.successful(sessionMissingRedirect()))
           }
         )
     }
   }
 
+  val showLinkAccount: Action[AnyContent] = Action.async { implicit request =>
+    withAuthenticatedAgent {
+      Future.successful {
+        request.session.get("arn").fold(sessionMissingRedirect()) { _ =>
+          Ok(html.link_account(linkAccountForm))
+        }
+      }
+    }
+  }
+
+  val submitLinkAccount: Action[AnyContent] = Action.async { implicit request =>
+    withAuthenticatedAgent {
+      request.session.get("arn").fold(Future.successful(sessionMissingRedirect())) { arn =>
+        linkAccountForm
+          .bindFromRequest()
+          .fold(
+            formWithErrors => {
+              if (formWithErrors.errors.exists(_.message == "error.link-account-value.invalid")) {
```

Comment:
Not sure `error.link-account-value.invalid` has been added in the messages conf.

Reply:
This is the case where the user has been sending up values in the request that aren't even possible from the rendered HTML form.